### PR TITLE
[feature] Add support for Artifactory projects in build info commands

### LINKED ISSUE #38: Add projects support to art:build-info commands
Trying to organize and prep a demo, I wants to use a Project in Artifactory to separate my artifacts and builds.

image

How the URLs are missing the query parm for this https://jfrog.com/help/r/jfrog-rest-apis/build-promotion `GET /api/build?project='projectKey'`

### PR DESCRIPTION
**Pull request by danimtb**
Includes work at https://github.com/conan-io/conan-extensions/pull/56

- Added the `--project` flag for upload, append, get, promote & delete commands.
- Added complete test with all the commands using the new flag.

closes #38
closes #56
closes #39

### CODE REVIEW

File: extensions/commands/art/cmd_build_info.py [hunk 1]
```diff
@@ -319,11 +321,14 @@ def _check_min_required_conan_version(min_ver):
                              f"Conan versions>={min_ver}. Please update Conan.")
 
 
-def _add_default_arguments(subparser, is_bi_create=False):
+def _add_default_arguments(subparser, is_bi_create=False, is_bi_create_bundle=False):
     url_help = "Artifactory url, like: https://<address>/artifactory."
     if is_bi_create:
         url_help += " This may be not necessary if all the information for the Conan artifacts is present in the " \
                     "local cache."
+    if not (is_bi_create_bundle or is_bi_create_bundle):
```
Comment (czoido) [inline extensions/commands/art/cmd_build_info.py:329]:
```suggestion
    if not is_bi_create_bundle:
```

**Review by czoido — APPROVED** [review]
LGTM, the only thing missing is adding the documentation to the README

File: extensions/commands/art/cmd_build_info.py (hunk 1 above)
Comment (danimtb) [reply extensions/commands/art/cmd_build_info.py:329]: The `--project` flag does not apply to the bi-create neither to bi-create-bundle, so we have to keep that

**Comment by danimtb** [comment]
> LGTM, the only thing missing is adding the documentation to the README

I took a look at the readme but can see a good place for it. Also, my thought was that this new `--project` flag is optional, and I didn't want to add more things to the commands of the example. If you think we should add it somewhere, please open an issue and I'll be happy to contribute it in a PR 😃

File: extensions/commands/art/cmd_build_info.py [hunk 2]
```diff
@@ -497,9 +505,12 @@ def build_info_delete(conan_api: ConanAPI, parser, subparser, *args):
         "deleteArtifacts": "true" if args.delete_artifacts else "false",
         "deleteAll": "true" if args.delete_all else "false",
     }
+    # Deleting BIs from projets is not documented in the artifactory api: https://jfrog.com/help/r/jfrog-rest-apis/delete-builds
+    # the usual url parameter ?project={args.project} does not work here
```
Comment (prince-chrismc) [inline extensions/commands/art/cmd_build_info.py:509]: 🤔 Maybe we should open a ticket or ask on slack?

**Pull request merged**